### PR TITLE
Reader: Add ABTest for search on following

### PR DESCRIPTION
**Pull request by blowery**
This adds a new ABTest that shows the search box at the top of the following stream. To test it, enable the 'show' variation of the readerSearchOnFollowing abtest.

### CODE REVIEW

File: client/lib/abtest/active-tests.js [hunk 1]
```diff
@@ -157,5 +157,15 @@ module.exports = {
...
 		allowExistingUsers: true
+	},
+
+	readerSearchOnFollowing: {
+		datestamp: '20170206',
+		variations: {
+			show: 50,
+			hide: 50
+		},
+		defaultVariation: 'hide',
```
Comment (samouri) [inline client/lib/abtest/active-tests.js]: what does default do?

**Comment by samouri** [comment]
This looks awesome!

Can we remove the padding above the reader__content for only when search exists?
Otherwise there is a bit too much space above the content:

**Comment by samouri** [comment]
Tested:
- [x] searching from following works
- [x] metrics worked as expected (source=following for suggestions + `calypso_reader_search_from_following`)

Question:
any particular reason to use source to differentiate for suggestions but then make a new stat for tracking the search? (instead of using source for both)

File: client/lib/abtest/active-tests.js (hunk 1 above)
Comment (aaronyan) [inline client/lib/abtest/active-tests.js]: The `defaultVariation` is used when a user is not eligible to participate in the A/B test.

**Comment by blowery** [comment]
> any particular reason to use source to differentiate for suggestions but then make a new stat for tracking the search? (instead of using source for both)

The longer analytics flow for search is ... weird. 

So imagine you type in `pan` on the following stream, then wait a touch, and we bump you to `/read/search?q=pan`. We start loading results, and you type `ts` for `pants`. Would that be one search with source `following` and one with source `search`? Or two with `following`?

File: client/lib/abtest/active-tests.js [hunk 2]
```diff
@@ -157,5 +157,15 @@ module.exports = {
...
+		allowExistingUsers: true
 	}
```
Comment (aaronyan) [inline client/lib/abtest/active-tests.js]: Do we need to add `allowAnyLocale: true` to allow any language user to be admitted to the A/B test?

**Pull request merged**